### PR TITLE
Add test for inline functions

### PR DESCRIPTION
**Pull request by guyco33**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

relates to https://github.com/trinodb/trino/issues/19820

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/sql/query/TestInlineFunctions.java [hunk 1]
```diff
@@ -61,34 +61,45 @@ public void testInlineFunction()
...
                 .matches("SELECT nationkey * 2 FROM nation WHERE nationkey >= 1");
 
         assertThat(assertions.query(
                 """
-                WITH FUNCTION my_func(x bigint) 
-                    RETURNS bigint 
+                WITH FUNCTION my_func(x bigint)
+                    RETURNS bigint
                     RETURN x * 2
-                SELECT my_func(nationkey)
```
Comment (martint) [inline core/trino-main/src/test/java/io/trino/sql/query/TestInlineFunctions.java:89]: Move these formatting changes to a separate commit.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/TestFunctions.java [hunk 2]
```diff
@@ -45,4 +45,20 @@ public void testVersion()
                 onTrino().executeQuery("SELECT version()").getOnlyValue(),
                 onTrino().executeQuery("SELECT node_version FROM system.runtime.nodes WHERE coordinator = TRUE").getOnlyValue());
     }
+
+    @Test(groups = FUNCTIONS)
```
Comment (martint) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/TestFunctions.java:49]: We don't need a product test for this. It's covered in the engine tests above.

Product tests are more expensive to run and harder to work with, so they should only be limited to testing full-system integrations that cannot be replicated within unit tests or with testcontainers.

Comment (electrum) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/TestFunctions.java:49]: Also note that the existing engine tests effectively run on multiple nodes via DistributedQueryRunner.

Comment (guyco33) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/TestFunctions.java:49]: What would be the quickest solution to make those engine tests run via DistributedQueryRunner ?

**Comment by electrum** [comment]
I don't think we need to test with both `LocalQueryRunner` and `DistributedQueryRunner`. We should convert the existing tests to use `DistributedQueryRunner` if necessary.

**Comment by mosabua** [comment]
👋 @guyco33 and @electrum .. just a small nudge to maybe revisit this and merge..

**Comment by mosabua** [comment]
@guyco33 @martint @electrum .. do we want to close this or continue work?
